File: core/src/main/java/com/uiptv/model/PublishedM3uChannelSelection.java
package com.uiptv.model;

import com.uiptv.shared.BaseJson;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class PublishedM3uChannelSelection extends BaseJson {
    private String dbId;
    private String accountId;
    private String categoryName;
    private String channelId;
    private boolean selected;

    public PublishedM3uChannelSelection(String accountId, String categoryName, String channelId, boolean selected) {
        this.accountId = accountId;
        this.categoryName = categoryName;
        this.channelId = channelId;
        this.selected = selected;
    }
}

File: server/src/main/kotlin/com/uiptv/model/PublishedM3uChannelSelection.kt
package com.uiptv.model

import com.uiptv.shared.BaseJson

data class PublishedM3uChannelSelection @JvmOverloads constructor(
    var dbId: String? = null,
    var accountId: String? = null,
    var categoryName: String? = null,
    var channelId: String? = null,
    @get:JvmName("isSelected")
    var selected: Boolean = false
) : BaseJson() {
    constructor(accountId: String?, categoryName: String?, channelId: String?, selected: Boolean) : this(null, accountId, categoryName, channelId, selected)
}
